components/calendar/src/calendar-input/calendar-input.js

```diff
@@ -76,14 +70,26 @@ export const CalendarInput = ({
 
     return (
         <>
-            <div ref={ref}>
+            <div className="calendar-input-wrapper" ref={ref}>
                 <InputField
                     label="Pick a date"
                     {...rest}
                     type="text"
                     onFocus={onFocus}
                     value={value}
                 />
+                {clearable && (
+                    <div className="calendar-clear-button">
+                        <Button
+                            dataTest="storybook-clear-button"
```

Comment:
I'd recommend a name other than `storybook-clear-button` (as it's in the main component, rather than in a storybook story)